cmd/ingester/app/builder/builder.go

```diff
@@ -81,7 +80,6 @@ func CreateConsumer(logger *zap.Logger, metricsFactory metrics.Factory, spanWrit
 		ProcessorFactory:      *processorFactory,
 		MetricsFactory:        metricsFactory,
 		Logger:                logger,
-		DeadlockCheckInterval: options.DeadlockInterval,
```

Comment:
IIRC, this can be configured via CLI. You might need to mark it as "deprecated" in the CLI help text + add an entry to the changelog.